pkgs/tools/audio/beets/default.nix

```diff
@@ -17,11 +18,25 @@
 **   alternatives = { enable = true; propagatedBuildInputs = [ beetsPackages.alternatives ]; };
 ** }; }
 */
-lib.makeExtensible (self: {
+let
+  legacyMediafilePython3Packages = python3Packages.override {
+    overrides = self: super: {
+      mediafile = super.mediafile.overridePythonAttrs (oldAttrs: rec {
+        version = "0.10.1";
+        format = "pyproject";
+        src = fetchPypi {
+          pname = "mediafile";
+          inherit version;
+          hash = "sha256-kpZCoX7lAjuQhiIc6AzcLFHQYCGokNRDOwvVvTLysp8=";
+        };
+      });
+    };
+  };
```

Comment:
From what I could tell:
1. This is how to best use an older version of a package
2. This is kosher and there is precedent for pinning to an older version in the nixpkgs repo

If I am mistaken in either of these, such as if there's a way to depend on an older version of the nixpkg itself (rather than getting it from pypi), please let me know. I was having some trouble finding clear guidance in docs on this case.